### PR TITLE
Rename Vertex Editor dialog to Node Editor (fixes #17807)

### PR DESCRIPTION
**Pull request by DelazJ**
because the context menu proposes "Node editor" and also of the "Node Tool".

Actually I don't quite know if we should go for "Vertex" or "Node" in this context (but we should harmonize the context menu text with the dialog title). ~~I chose to keep node as it was the label in 2.x~~(I checked once again and it was actually vertex editor in 2.18)

### CODE REVIEW

**Comment by nirvn** [comment]
I have some reservation. I'm actually wondering whether we should go the other way around and rename node tool to vertex tool, etc.

ATM, we use both (for e.g., snapping configuration refers to vertex and segment while processing refers to nodes), it'd be nice to settle on one.

**Comment by DelazJ** [comment]
> I have some reservation. I'm actually wondering whether we should go the other way around 

Yep, was also what i wondered once I pushed the PR. Agreed to mimic vocabulary of snapping option but I'm not sure we should rename the node tool ("node" imho has some idea of topology which is worth keeping when reshaping geometries. And vertex does not.)

**Comment by nirvn** [comment]
@wonder-sk , @m-kuhn , @nyalldawson , @NathanW2 , others , what's your opinion on this? Should we settle on the use of one word across the board? If the answer is positive, _node_ or _vertex_?

**Comment by wonder-sk** [comment]
I do not have preference of one term over the other, and not being native speaker I don't know which one sounds better. I agree though with the idea that the naming should be consistent.

**Comment by nirvn** [comment]
My vote goes to _vertex_. While it might be less friendly of approach than _node_, we're in the world of geometry and IMHO we gain using the proper terminology on the long run.

When we settle on this, I'll be more than happy to update the processing algorithms referring to _node_... err, that is, if _vertex_ wins 😉

**Comment by nyalldawson** [comment]
My preference would be vertex.

**Comment by jmarsac** [comment]
Hi,
I agree with @nirvn and @DelazJ ; since "node" refers to topology, it's better to use "vertex"

**Comment by DelazJ** [comment]
Replacement done: node tool -> vertex tool and node editor -> vertex editor

**Comment by nirvn** [comment]
Big +1.

**Comment by nirvn** [comment]
I'd say yes. @wonder-sk , what's your opinion on the source file rename?

**Comment by wonder-sk** [comment]
Yeah it would be good to rename the source files and the class name for consistency

**Comment by nirvn** [comment]
@DelazJ , thanks for pushing this through the finish line. I'll merge as soon as travis turns green.

File: src/app/vertextool/qgsvertexeditor.cpp [hunk 1]
```diff
@@ -287,7 +287,7 @@ QgsNodeEditor::QgsNodeEditor(
   mCanvas = canvas;
 
   mTableView = new QTableView( this );
-  mNodeModel = new QgsNodeEditorModel( mLayer, mSelectedFeature, mCanvas, this );
+  mNodeModel = new QgsVertexEditorModel( mLayer, mSelectedFeature, mCanvas, this );
```
Comment (nirvn) [inline src/app/vertextool/qgsvertexeditor.cpp]: @DelazJ , mNodeModel -> mVertexModel?

File: src/app/vertextool/qgsvertexeditor.cpp [hunk 2]
```diff
@@ -300,11 +300,11 @@ QgsNodeEditor::QgsNodeEditor(
 
   setWidget( mTableView );
 
-  connect( mSelectedFeature, &QgsSelectedFeature::selectionChanged, this, &QgsNodeEditor::updateTableSelection );
-  connect( mTableView->selectionModel(), &QItemSelectionModel::selectionChanged, this, &QgsNodeEditor::updateNodeSelection );
+  connect( mSelectedFeature, &QgsSelectedFeature::selectionChanged, this, &QgsVertexEditor::updateTableSelection );
+  connect( mTableView->selectionModel(), &QItemSelectionModel::selectionChanged, this, &QgsVertexEditor::updateNodeSelection );
```
Comment (nirvn) [inline src/app/vertextool/qgsvertexeditor.cpp]: updateNodeSelection -> updateVertexSelection

File: src/app/vertextool/qgsvertexeditor.cpp [hunk 3]
```diff
@@ -355,7 +355,7 @@ void QgsNodeEditor::updateNodeSelection( const QItemSelection &selected, const Q
   mUpdatingNodeSelection = false;
 }
 
-void QgsNodeEditor::zoomToNode( int idx )
+void QgsVertexEditor::zoomToNode( int idx )
```
Comment (nirvn) [inline src/app/vertextool/qgsvertexeditor.cpp]: zoomToNode -> zoomToVertex

File: src/app/vertextool/qgsvertextool.cpp [hunk 4]
```diff
@@ -1996,7 +1996,7 @@ void QgsNodeTool::validateGeometry( QgsVectorLayer *layer, QgsFeatureId featureI
   mValidations.insert( id, validation );
 }
 
-void QgsNodeTool::zoomToNode( const Vertex &node )
+void QgsVertexTool::zoomToNode( const Vertex &node )
```
Comment (nirvn) [inline src/app/vertextool/qgsvertextool.cpp]: zoomToNode -> zoomToVertex 
&node -> &vertex

**Pull request merged**